metadata/src/lib.rs

```diff
@@ -61,17 +59,20 @@ impl TryFrom<RuntimeMetadataPrefixed> for Metadata {
 		let mut modules_with_calls = HashMap::new();
 		for module in convert(meta.modules)?.into_iter() {
 			if let Some(calls_meta) = module.calls {
-				let mut calls = HashMap::new();
-				for (index, call) in convert(calls_meta)?.into_iter().enumerate() {
-					let call_name = convert(call.name)?;
-					calls.insert(call_name.to_string(), index as u8);
-				}
+				let calls = convert(calls_meta)?
+					.into_iter()
+					.enumerate()
+					.map(|(index, call)| {
+						let call_name = convert(call.name)?;
+						let index = u8::try_from(index)?;
+
+						Ok((call_name, index))
+					})
+					.collect::<Result<HashMap<String, u8>, substrate_airgapped::Error>>()?;
```

Comment:
Space complexity wise, using `.collect()`  here seems more efficient, but since we have to `.map()` first I would think the time complexity is worse by a constant.

Reply:
> Not sure I follow the space complexity argument, can you elaborate?

I think Zeke refers to that compiler can determine how much space to pre-allocate when using a `for_each.collect()` instead of creating an empty HashMap and push to it in every iteration.

> .map() first I would think the time complexity is worse by a constant.

No, you can think of `.map` as something you do in your for loop O(n), it most likely inlined and comes with no extra cost as most abstractions in Rust.

Reply:
> Having to turbo-fish the full return type is a bit noisy; you sure the type inference can't figure this out on its own?

Compiler complains if I take out the turbo fish